swan-lake/integration-technology-tools/edi-tool.md

```diff
@@ -6,4 +6,349 @@ keywords: ballerina, programming language, edi, contract
+
+The command below can be used to generate EDI libraries.
+
+```
+bal edi libgen -O <org name> -n <library name> -s <EDI schema folder> -o <output folder>
+```
+
+A Ballerina library project will be generated in the output folder. This library can be built and published by issuing the `bal pack` and `bal push` commands from the output folder. Then, the generated library can be imported into any Ballerina project and the generated utility functions of the library can be invoked to parse EDI messages into Ballerina records.
+
+For example, if an organization (`CityMart`) needs to work with `X12 850`, `810`, `820`, and `855` for handling purchase orders, then, its integration developers can put schemas of those `X12` specifications into a folder as follows.
+
+```
+|-- CityMart
+    |--lib
+    |--schemas
+       |--850.json
+       |--810.json
+       |--820.json
+       |--855.json
+```
+
+Execute the `libgen` command to generate a Ballerina library as shown below.
+
+```
+bal edi libgen -O citymart -n porder -s CityMart/schemas -o CityMart/lib
+```
+
+The generated Ballerina library will be as shown below.
+
+```ballerina
+|-- CityMart
+    |--lib  
+    |--porder
+    |     |--modules
+    |	  |   |--m850
+    |	  |	  |  |--G_850.bal
+    |     |   |  |--transformer.bal
+    |	  |	  |--m810
+    |	  |	  |  |--G_810.bal
+    |     |   |  |--transformer.bal
+    |	  |	  |--m820
+    |	  |	  |  |--G_820.bal
+    |     |   |  |--transformer.bal
+    |	  |	  |--m855
+    |	  |	    |--G_855.bal
+    |     |     |--transformer.bal
+    |	  |--Ballerina.toml
+    |	  |--Module.md
+    |	  |--Package.md
+    |	  |--porder.bal
+    |	  |--rest_connector.bal
+    |
+    |--schemas
+       |--850.json
+       |--810.json
+       |--820.json
+       |--855.json
+```
+
+As seen in the above project structure, the code for each EDI schema is generated into a separate module to prevent possible conflicts. Now, it is possible to build the above project using the `bal pack` command and publish it into [Ballerina Central](https://central.ballerina.io/) using the `bal push` command. 
+
+Then, any Ballerina project can import this package and use it to work with the EDI files related to purchase orders. An example of using this library for reading an `850` file and writing an `855` file is shown below.
+
+```ballerina
+import ballerina/io;
+import citymart/porder.m850;
+import citymart/porder.m855;
+
+public function main() returns error? {
+    string orderText = check io:fileReadString("orders/d15_05_2023/order10.edi");
+    m850:Purchase_Order purchaseOrder = check m850:fromEdiString(orderText);
+    ...
+    m855:Purchase_Order_Acknowledgement orderAck = {...};
+    string orderAckText = check m855:toEdiString(orderAck);
+    check io:fileWriteString("acks/d15_05_2023/ack10.edi", orderAckText);
+}
```

Comment:
Can we have a working code snippet here?

Reply:
Purchase_Order_Acknowledgement is a large record generated from a real EDI schema. It has 50+ fields, so not practical to show in the doc.

Reply:
Resolving as per the above comment.